controllers/cluster.weave.works/existinginframachine_controller.go

```diff
@@ -1023,20 +996,60 @@ func (a *ExistingInfraMachineReconciler) modifyNode(ctx context.Context, nodeNam
 	return nil
 }
 
+func (a *ExistingInfraMachineReconciler) modifyEIM(ctx context.Context, eim *existinginfrav1.ExistingInfraMachine, updater func(*existinginfrav1.ExistingInfraMachine)) error {
+	contextLog := log.WithFields(log.Fields{"eim": eim.Name})
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		var result existinginfrav1.ExistingInfraMachine
+		getErr := a.Client.Get(ctx, client.ObjectKey{Namespace: eim.Namespace, Name: eim.Name}, &result)
+		if getErr != nil {
+			contextLog.Errorf("failed to read machine info, assuming unsafe to update: %v", getErr)
```

Comment:
Nit - we are actually reading the existinginframachine instead of a machine.

Reply:
Sure